### ISSUE TITLE
Add attention layers

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/axon#271]
We should work out a common API and implementation for working with attention layers. Right now they are easily implemented as custom layers, but we should have conveniences to save user's time

**Comment by joaogui1** [comment]
Can I take this on?

**Comment by seanmor5** [comment]
Yes, definitely! Let me know if you run into any issues or have questions
